plugins/core-classes/src/main/java/org/qbicc/plugin/coreclasses/CoreClasses.java

```diff
@@ -248,6 +208,100 @@ private static FieldElement makeContentField(final int index, final DefinedTypeD
         return fieldBuilder.build();
     }
 
+    /**
+     * Initialize all of our fields ahead of time using patcher.
+     *
+     * @param ctxt the compilation context (must not be {@code null})
+     */
+    public static void init(CompilationContext ctxt) {
```

Comment:
Nice!  I hadn't thought about doing this programatically.  I was envisioning having to have a Object$_patch.java in the class library and dealing with all the complexity of keeping the class library and compiler in synch...

Reply:
The programmatic API was almost an afterthought TBH... but I think it might turn out to be very useful if we can fix the object layout *before* the interpreter runs.